dominode-bootstrapper/dominode_bootstrapper/geonodeadmin.py

```diff
@@ -175,20 +175,58 @@ def from_geonode_manager(cls, geonode_manager: GeoNodeManager):
         )
 
     def list_workspaces(self):
-        raise NotImplementedError  # TODO: Provide implementation
-
-    def create_workspace(self):
-        raise NotImplementedError  # TODO: Provide implementation
+        response = self.client.get(
+            f'{self.base_url}/rest/workspaces',
+            auth=(DEFAULT_USERNAME, DEFAULT_PASSWORD)
+        )
+        return response.json()
 
-    def create_postgis_store(self):
-        raise NotImplementedError  # TODO: Provide implementation
+    def create_workspace(self, name):
+        response = self.client.post(
+            f'{self.base_url}/rest/workspaces',
+            auth=(DEFAULT_USERNAME, DEFAULT_PASSWORD),
+            json={"name": name}
+        )
+        return response.json()
+    def get_workspace(self, name):
+        response = self.client.post(
+            f'{self.base_url}/rest/workspaces/{name}',
+            auth=(DEFAULT_USERNAME, DEFAULT_PASSWORD)
+        )
+        if response.status_code != 200:
+            return None
+        return response.json()
```

Comment:
Missing some newline here, between this method and the one above.

Personally I'd not bother with this method, since you can do `manager.list_workspaces()` and then just filter them out by name. The usage that we are doing of this is very limited, this method exists only to see if a workspace exists (check line #371 below). I'll leave that at your discretion.

Reply:
Thanks, I'm going to leave it as it is.